src/core_write.cpp

```diff
@@ -88,7 +88,12 @@ string ScriptToAsmStr(const CScript& script, const bool fAttemptSighashDecode)
         }
         if (0 <= opcode && opcode <= OP_PUSHDATA4) {
             if (vch.size() <= static_cast<vector<unsigned char>::size_type>(4)) {
-                str += strprintf("%d", CScriptNum(vch, false).getint());
+                if ( (lastopcode == OP_NAME_NEW || lastopcode == OP_NAME_UPDATE || lastopcode == OP_NAME_FIRSTUPDATE )
+                    && vch.size() > static_cast<vector<unsigned char>::size_type>(0) ) {
```

Comment:
Yes, if you don't mind, I'd appreciate the change.

I've never really used the `asm` encoding a lot myself, so I don't particularly care about it.  I don't know how 0-length data is represented.  But that is something that concerns also upstream Bitcoin, so should be fixed there (in case there's an issue with it).